### ISSUE TITLE
Kubernetes auth-realm causes Traefik to crash

### ISSUE THREAD

**Issue by rkistner** [containous/traefik#2239]
### Do you want to request a *feature* or report a *bug*?

bug

### What did you do?

1. Deploy traefik to a Kubernetes cluster, and set it up to handle ingress (everything good so far).
2. Configure ingress for a service (still good, everything working).
3. Add an `ingress.kubernetes.io/auth-realm` annotation to the ingress (see below).

### What did you expect to see?

The auth-realm should either be used, or ignored with a warning.

### What did you see instead?

Traefik crashes with error `Provider connection error: no realm customization supported`. This causes ingress for the entire cluster to be unavailable. The pod is automatically restarted, but it continues crashing, until the `auth-realm` annotation is removed.

### Output of `traefik version`: (_What version of Traefik are you using?_)

traefik:v1.3.8

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Deployed via helm, with fairly standard config: https://github.com/kubernetes/charts/tree/master/stable/traefik

Ingress configuration looks like this:

```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: kube-dashboard
  namespace: kube-system
  annotations:
    kubernetes.io/ingress.class: "traefik"
    ingress.kubernetes.io/auth-type: basic
    ingress.kubernetes.io/auth-realm: "Authenticate" # Problematic line
    ingress.kubernetes.io/auth-secret: admin-htpasswd
spec:
  rules:
  - host: dashboard.my.cluster
    http:
      paths:
      - path: /
        backend:
          serviceName: kubernetes-dashboard
          servicePort: 80
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
time="2017-10-10T10:24:56Z" level=info msg="Traefik version v1.3.8 built on 2017-09-07_08:46:19PM" 
time="2017-10-10T10:24:56Z" level=info msg="Using TOML configuration file /config/traefik.toml" 
time="2017-10-10T10:24:56Z" level=info msg="Preparing server https &{Network: Address::443 TLS:0xc420048a80 Redirect:<nil> Auth:<nil> Compress:true}" 
time="2017-10-10T10:25:00Z" level=info msg="Loading ACME Account..." 
time="2017-10-10T10:25:00Z" level=info msg="Loaded ACME config from store /acme/acme.json" 
time="2017-10-10T10:25:01Z" level=info msg="Preparing server http &{Network: Address::80 TLS:<nil> Redirect:<nil> Auth:<nil> Compress:true}" 
time="2017-10-10T10:25:01Z" level=info msg="Starting provider *server.WebProvider {"Address":":8080","CertFile":"","KeyFile":"","ReadOnly":false,"Statistics":null,"Metrics":null,"Path":"","Auth":{"Basic":{"Users":["admin:<redacted>"],"UsersFile":""},"Digest":null,"HeaderField":""}}" 
time="2017-10-10T10:25:01Z" level=info msg="Starting provider *kubernetes.Provider {"Watch":true,"Filename":"","Constraints":[],"Endpoint":"","Token":"","CertAuthFilePath":"","DisablePassHostHeaders":false,"Namespaces":null,"LabelSelector":""}" 
time="2017-10-10T10:25:01Z" level=info msg="Retrieving ACME certificates..." 
time="2017-10-10T10:25:01Z" level=info msg="Retrieved ACME certificates" 
time="2017-10-10T10:25:01Z" level=info msg="Starting server on :443" 
time="2017-10-10T10:25:01Z" level=info msg="Starting server on :80" 
time="2017-10-10T10:25:01Z" level=info msg="Creating in-cluster Provider client
" 
time="2017-10-10T10:25:02Z" level=error msg="Provider connection error: no realm customization supported; retrying in 267.639422ms" 
ERROR: logging before flag.Parse: E1010 10:25:02.235691       1 runtime.go:66] Observed a panic: "send on closed channel" (send on closed channel)
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:72
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:65
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/usr/local/go/src/runtime/chan.go:180
/usr/local/go/src/runtime/chan.go:604
/go/src/github.com/containous/traefik/provider/kubernetes/client.go:136
/go/src/github.com/containous/traefik/provider/kubernetes/client.go:143
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/tools/cache/controller.go:173
<autogenerated>:50
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/tools/cache/controller.go:251
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:420
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/tools/cache/controller.go:131
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/tools/cache/controller.go:102
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/wait/wait.go:96
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/wait/wait.go:97
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/wait/wait.go:52
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/tools/cache/controller.go:102
/usr/local/go/src/runtime/asm_amd64.s:2197
ERROR: logging before flag.Parse: E1010 10:25:02.235872       1 runtime.go:66] Observed a panic: "send on closed channel" (send on closed channel)
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:72
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:65
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/go/src/github.com/containous/traefik/vendor/k8s.io/client-go/pkg/util/runtime/runtime.go:58
/usr/local/go/src/runtime/asm_amd64.s:514
/usr/local/go/src/runtime/panic.go:489
/usr/local/go/src/runtime/chan.go:180
/usr/local/go/src/runtime/chan.go:604
```

**Comment by ldez** [comment]
Could you try like that:

```yml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: kube-dashboard
  namespace: kube-system
  annotations:
    kubernetes.io/ingress.class: "traefik"
    ingress.kubernetes.io/auth-type: basic
    ingress.kubernetes.io/auth-realm: "traefik" # <---------------
    ingress.kubernetes.io/auth-secret: admin-htpasswd
spec:
  rules:
  - host: dashboard.my.cluster
    http:
      paths:
      - path: /
        backend:
          serviceName: kubernetes-dashboard
          servicePort: 80
```

**Comment by rkistner** [comment]
Yes, setting it to `"traefik"`, or not setting it at all, does work.

I could not find any official Kubernetes documentation on what this setting is supposed to do. However, some nginx examples use this for "message to display with an appropiate context why the authentication is required" (see https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx/examples/auth).

However, for me the big issue is not what exactly this does or the what the value is supposed to be. The issue is that by misconfiguring a single service, ingress for the entire cluster can go down. If is was just the single service that was affected, this would be much less of an issue.

**Comment by timoreimann** [comment]
Definitely a bug. We should tackle it.

**Comment by ldez** [comment]
Closed by #2239
